Small refactor by fixing clippy warnings

ValentinLeTallec opened the pull request.

jnioche commented on cloud-scanner-cli/src/boavizta_api_v1.rs:
```diff
@@ -357,7 +357,7 @@ mod tests {
         let api: BoaviztaApiV1 = BoaviztaApiV1::new(TEST_API_URL);
         let version = api.get_api_version().await;
-        let expected = Some("1.3.7".to_owned());
+        let expected = Some("1.3.8".to_owned());
```
Let's do the version upgrade separately across the whole project. 
https://github.com/Boavizta/cloud-scanner/blob/main/docker-compose.yml#L60 will need changing as well

demeringo replied:
Registered an issue to track this upgrade separately.
https://github.com/Boavizta/cloud-scanner/issues/681

jnioche reviewed (changes requested):
Thanks @ValentinLeTallec 
Could you add a line to the CHANGES files as part of this PR?
See comment about version change
I can't see any clippy warnings on https://github.com/Boavizta/cloud-scanner/actions/runs/13189896898/job/36820573288
did you get them by running it on the command line? if so, should we do similar warnings  as part of our actions?

demeringo commented:
> Thanks @ValentinLeTallec Could you add a line to the CHANGES files as part of this PR? See comment about version change I can't see any clippy warnings on https://github.com/Boavizta/cloud-scanner/actions/runs/13189896898/job/36820573288 did you get them by running it on the command line? if so, should we do similar warnings as part of our actions?

Clippy warning may also differ accoring to the version of the Rust toolset.

@ValentinLeTallec could you please provide the output of `rustup --version` (if you use Rustup to setup Rust of course).

On the laptop I'm typing this right now I have a slightly older version of the compiler (1.82), maybe this explains the difference.
```sh
rustup --version
rustup 1.27.1 (54dd3d00f 2024-04-24)
info: This is the version for the rustup toolchain manager, not the rustc compiler.
info: The currently active `rustc` version is `rustc 1.82.0 (f6e511eec 2024-10-15)`


cargo clippy --version
clippy 0.1.82 (f6e511e 2024-10-15)
```

ValentinLeTallec commented:
Indeed, I am on `1.83` 

```
❯  cargo --version
cargo 1.83.0 (5ffbef321 2024-10-29)

❯ cargo clippy --version
clippy 0.1.83

❯ rustc --version
rustc 1.83.0 (90b35a623 2024-11-26) (built from a source tarball) 
```
Thought to be more specific, the warning from default clippy that I have fixed are in 302e83887b6ece29dbaa25078d80321e7b197503.
For f56c2e214207f051ebe4547d76c6f0ad05dbcd3d ea715b8f5654e2218dc51ff27e37f1aa43aa95d3 I added some warnings that I find useful by add the following to the top of  `cloud-scanner-lambda/src/main.rs` and `cloud-scanner-cli/src/main.rs`.

```rust
#![warn(
    clippy::all,
    clippy::pedantic,
    clippy::nursery,
    clippy::cargo,
    clippy::unwrap_used,
    clippy::expect_used,
    clippy::wildcard_enum_match_arm,
    clippy::use_debug
)]
#![allow(clippy::multiple_crate_versions)] // This one comes from `clippy::cargo` but is hard to resolve so I selectively allow it.
```
